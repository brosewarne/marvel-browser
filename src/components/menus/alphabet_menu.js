import React from 'react';
import { Link } from 'react-router-dom';

import './menu.css';

/**
 * Render the AlphabetMenu - used for selecting a new `startsWith` letter.
 * @returns {Node} - The rendered AlphabetMenu
 * @constructor
 */
export const AlphabetMenu = () => {
    const alphabet = '3ABCDEFGHIJKLMNOPQRSTUVWXYZ'.split('');

    const alphabetTiles = alphabet.map((letter) => {
        const href = `/characters/${letter}/1`;
        return (
            <div className="menuTile" key={`alphabet_${letter}_div`}>
                <Link href={href} to={href} key={`alphabet_${letter}`}>{letter}</Link>
            </div>
        );
    });
    return (
        <div className="menu">
            {alphabetTiles}
        </div>
    );
};
