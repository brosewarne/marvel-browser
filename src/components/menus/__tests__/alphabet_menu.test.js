import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import { AlphabetMenu } from '../alphabet_menu';

configure({ adapter: new Adapter() });

describe('<AlphabetMenu />', () => {
    it('renders correctly', () => {
        const wrapper = shallow(
            <AlphabetMenu />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
