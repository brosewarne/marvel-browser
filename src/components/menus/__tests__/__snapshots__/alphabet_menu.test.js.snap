// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlphabetMenu /> renders correctly 1`] = `
<div
  className="menu"
>
  <div
    className="menuTile"
    key="alphabet_3_div"
  >
    <Link
      href="/characters/3/1"
      key="alphabet_3"
      replace={false}
      to="/characters/3/1"
    >
      3
    </Link>
  </div>
  <div
    className="menuTile"
    key="alphabet_A_div"
  >
    <Link
      href="/characters/A/1"
      key="alphabet_A"
      replace={false}
      to="/characters/A/1"
    >
      A
    </Link>
  </div>
  <div
    className="menuTile"
    key="alphabet_B_div"
  >
    <Link
      href="/characters/B/1"
      key="alphabet_B"
      replace={false}
      to="/characters/B/1"
    >
      B
    </Link>
  </div>
  <div
    className="menuTile"
    key="alphabet_C_div"
  >
    <Link
      href="/characters/C/1"
      key="alphabet_C"
      replace={false}
      to="/characters/C/1"
    >
      C
    </Link>
  </div>
  <div
    className="menuTile"
    key="alphabet_D_div"
  >
    <Link
      href="/characters/D/1"
      key="alphabet_D"
      replace={false}
      to="/characters/D/1"
    >
      D
    </Link>
  </div>
  <div
    className="menuTile"
    key="alphabet_E_div"
  >
    <Link
      href="/characters/E/1"
      key="alphabet_E"
      replace={false}
      to="/characters/E/1"
    >
      E
    </Link>
  </div>
  <div
    className="menuTile"
    key="alphabet_F_div"
  >
    <Link
      href="/characters/F/1"
      key="alphabet_F"
      replace={false}
      to="/characters/F/1"
    >
      F
    </Link>
  </div>
  <div
    className="menuTile"
    key="alphabet_G_div"
  >
    <Link
      href="/characters/G/1"
      key="alphabet_G"
      replace={false}
      to="/characters/G/1"
    >
      G
    </Link>
  </div>
  <div
    className="menuTile"
    key="alphabet_H_div"
  >
    <Link
      href="/characters/H/1"
      key="alphabet_H"
      replace={false}
      to="/characters/H/1"
    >
      H
    </Link>
  </div>
  <div
    className="menuTile"
    key="alphabet_I_div"
  >
    <Link
      href="/characters/I/1"
      key="alphabet_I"
      replace={false}
      to="/characters/I/1"
    >
      I
    </Link>
  </div>
  <div
    className="menuTile"
    key="alphabet_J_div"
  >
    <Link
      href="/characters/J/1"
      key="alphabet_J"
      replace={false}
      to="/characters/J/1"
    >
      J
    </Link>
  </div>
  <div
    className="menuTile"
    key="alphabet_K_div"
  >
    <Link
      href="/characters/K/1"
      key="alphabet_K"
      replace={false}
      to="/characters/K/1"
    >
      K
    </Link>
  </div>
  <div
    className="menuTile"
    key="alphabet_L_div"
  >
    <Link
      href="/characters/L/1"
      key="alphabet_L"
      replace={false}
      to="/characters/L/1"
    >
      L
    </Link>
  </div>
  <div
    className="menuTile"
    key="alphabet_M_div"
  >
    <Link
      href="/characters/M/1"
      key="alphabet_M"
      replace={false}
      to="/characters/M/1"
    >
      M
    </Link>
  </div>
  <div
    className="menuTile"
    key="alphabet_N_div"
  >
    <Link
      href="/characters/N/1"
      key="alphabet_N"
      replace={false}
      to="/characters/N/1"
    >
      N
    </Link>
  </div>
  <div
    className="menuTile"
    key="alphabet_O_div"
  >
    <Link
      href="/characters/O/1"
      key="alphabet_O"
      replace={false}
      to="/characters/O/1"
    >
      O
    </Link>
  </div>
  <div
    className="menuTile"
    key="alphabet_P_div"
  >
    <Link
      href="/characters/P/1"
      key="alphabet_P"
      replace={false}
      to="/characters/P/1"
    >
      P
    </Link>
  </div>
  <div
    className="menuTile"
    key="alphabet_Q_div"
  >
    <Link
      href="/characters/Q/1"
      key="alphabet_Q"
      replace={false}
      to="/characters/Q/1"
    >
      Q
    </Link>
  </div>
  <div
    className="menuTile"
    key="alphabet_R_div"
  >
    <Link
      href="/characters/R/1"
      key="alphabet_R"
      replace={false}
      to="/characters/R/1"
    >
      R
    </Link>
  </div>
  <div
    className="menuTile"
    key="alphabet_S_div"
  >
    <Link
      href="/characters/S/1"
      key="alphabet_S"
      replace={false}
      to="/characters/S/1"
    >
      S
    </Link>
  </div>
  <div
    className="menuTile"
    key="alphabet_T_div"
  >
    <Link
      href="/characters/T/1"
      key="alphabet_T"
      replace={false}
      to="/characters/T/1"
    >
      T
    </Link>
  </div>
  <div
    className="menuTile"
    key="alphabet_U_div"
  >
    <Link
      href="/characters/U/1"
      key="alphabet_U"
      replace={false}
      to="/characters/U/1"
    >
      U
    </Link>
  </div>
  <div
    className="menuTile"
    key="alphabet_V_div"
  >
    <Link
      href="/characters/V/1"
      key="alphabet_V"
      replace={false}
      to="/characters/V/1"
    >
      V
    </Link>
  </div>
  <div
    className="menuTile"
    key="alphabet_W_div"
  >
    <Link
      href="/characters/W/1"
      key="alphabet_W"
      replace={false}
      to="/characters/W/1"
    >
      W
    </Link>
  </div>
  <div
    className="menuTile"
    key="alphabet_X_div"
  >
    <Link
      href="/characters/X/1"
      key="alphabet_X"
      replace={false}
      to="/characters/X/1"
    >
      X
    </Link>
  </div>
  <div
    className="menuTile"
    key="alphabet_Y_div"
  >
    <Link
      href="/characters/Y/1"
      key="alphabet_Y"
      replace={false}
      to="/characters/Y/1"
    >
      Y
    </Link>
  </div>
  <div
    className="menuTile"
    key="alphabet_Z_div"
  >
    <Link
      href="/characters/Z/1"
      key="alphabet_Z"
      replace={false}
      to="/characters/Z/1"
    >
      Z
    </Link>
  </div>
</div>
`;
